### PR TITLE
Initial support for account entitlements

### PR DESCRIPTION
**Pull request by bhavyaus**
This pr does not enable this feature.

### CODE REVIEW

File: src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		this.isInitialized = true;
+
+		const init = {
+			method: 'GET',
+			headers: new Headers({
+				'Authorization': `Bearer ${session.accessToken}`
+			}),
+		};
+
+		const response = await fetch(this.productService.entitlement!.entitlementUrl, init);
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts:116]: use the `RequestService` for this

File: src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+			return;
+		}
+		const parsedResult = await response.json();
+		if (!parsedResult) {
+			return;
+		}
+
+		const accountsMenuBadgeDisposable = this._register(new MutableDisposable());
+
+		const orgs = parsedResult['organization_login_list'] as any[];
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts:127]: copilot specific stuff?

File: src/vs/base/common/product.ts [hunk 3]
```diff
@@ -188,6 +188,7 @@ export interface IProductConfiguration {
 
 	readonly commonlyUsedSettings?: string[];
 	readonly aiGeneratedWorkspaceTrust?: IAiGeneratedWorkspaceTrust;
+	readonly entitlement?: IEntitlement;
```
Comment (TylerLeonhardt) [inline src/vs/base/common/product.ts:191]: only a single entitlement?

File: src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts (hunk 1 above)
Comment (bhavyaus) [reply src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts]: Done.

File: src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts (hunk 2 above)
Comment (bhavyaus) [reply src/vs/workbench/contrib/accountEntitlements/browser/accountsEntitlements.contribution.ts]: Not really. Entitlements are tied to orgs, I decided to use it as is.

File: src/vs/base/common/product.ts (hunk 3 above)
Comment (bhavyaus) [reply src/vs/base/common/product.ts:191]: For now yes. I thought about enabling this for multiple entitlements but those APIs could differ and not have the same response shape. I'll enable this for now and if we do decide to take this route, we should look towards standardizing it.

File: src/vs/base/common/product.ts [hunk 4]
```diff
@@ -288,3 +289,12 @@ export interface IAiGeneratedWorkspaceTrust {
 	readonly dontTrustOption: string;
 	readonly startupTrustRequestLearnMore: string;
 }
+
+export interface IEntitlement {
```
Comment (TylerLeonhardt) [inline src/vs/base/common/product.ts:293]: This seems GitHub specific. Can you rename this and the property in the product.json

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**